integreat_cms/static/src/js/media-management/component/directory-entry.tsx

```diff
@@ -10,21 +12,53 @@ interface Props {
   onClick?: (event: MouseEvent) => void;
   mediaTranslations: any;
   globalEdit?: boolean;
+  allowDrop: boolean;
+  itemDropped: () => unknown;
+  dragStart: () => unknown;
+  dragEnd: () => unknown;
 }
 
 export default function DirectoryEntry({
   directory,
   onClick,
   mediaTranslations,
   globalEdit,
+  allowDrop,
+  itemDropped: fileDropped,
+  dragStart, dragEnd
 }: Props) {
+  const [isCurrentDropTarget, setIsCurrentDropTarget] = useState(false);
+
   return (
     <div
       title={mediaTranslations.btn_enter_directory}
       className={
-        "relative cursor-pointer hover:text-blue-500 flex flex-col justify-between h-full p-2"
+        cn("relative cursor-pointer hover:text-blue-500 flex flex-col justify-between h-full p-2",
+        {"text-blue-500": isCurrentDropTarget})
       }
       onClick={onClick}
+      onDragOver={(e) => {
+        if(allowDrop) {
+          e.preventDefault();
+          setIsCurrentDropTarget(true);
+        }
+      }}
+      onDragLeave={(e) => {
+        if(allowDrop) {
+          e.preventDefault();
+          setIsCurrentDropTarget(false);
+        }
+      }}
+      onDrop={(e) => {
+        if(allowDrop) {
+          e.preventDefault();
+          setIsCurrentDropTarget(false);
+          fileDropped();
+        }
+      }}
+      onDragStart={dragStart}
+      onDragEnd={dragEnd}
+      draggable={!allowDrop && (!directory.isGlobal || globalEdit)}
     >
       <Folder className={"w-full h-24 flex-none"} />
```

Comment:
```suggestion
      {isCurrentDropTarget ? (
        <FolderOpen className={"w-full h-24 flex-none"} />
      ) : (
        <Folder className={"w-full h-24 flex-none"} />
      )}
```
We could open the folder on hover...